### DISCUSSION TITLE
Arch module to know what if terminal is emulated by rosetta on M1

### DISCUSSION THREAD

**Discussion by zalewskigrzegorz** [starship/starship#2385] [Ideas]
I got an idea, and it helps me a lot but don't know how to configure it in a starship.

Currently, I need to switch between arm64 and i386 arch on my daily tasks, and I forget what version I have in my current window. There is a way to configure the arch module to show the icon when the terminal is in i386 emulation mode?
It's easy to check by arch command.

**Comment by matchai** [comment]
In the meantime, here's a quick solution using environment variable support: https://gist.github.com/iansu/1ad9fc5db90aafbf14ce58e99d5940ef

**Reply by zalewskigrzegorz** [reply]
works for fish with this in `~/.config/fish/config.fish` : 
```
set -x SHELL_ARCH (arch)
```
